test/libsolidity/SolidityScanner.cpp

```diff
@@ -622,9 +638,9 @@ BOOST_AUTO_TEST_CASE(irregular_line_breaks_in_single_line_doc_comment)
 
 BOOST_AUTO_TEST_CASE(regular_line_breaks_in_strings)
 {
-	for (auto const& nl: {"\n", "\r"})
+	for (auto const& nl: {"\r"s, "\n"s, "\r\n"s})
```

Comment:
Ok, this suggests that we did and still do consider stray ``\r`` valid line breaks...

Reply:
Still a bit undecided on allowing CR only as a valid line break. I'd try to follow what modern programming languages do, but they behave differently (starting point of my research was this Rust issue: https://github.com/rust-lang/rust/issues/7946). 

Maybe let's keep it for now and discuss again?

Reply:
The safest thing might be to actually throw on single ``\r``, but yeah - we can discuss this separately.

Reply:
I found out that some languages treat stray`\r` as whitespace, which seem to simplify parsing on their side. Adopting this behaviour would also affect this PR positively ;)